linode/terraform-provider-linode
# Allow wait for replications and update replica_regions requirements

## 📝 Description

Add a new attribute `wait_for_replications` allow user to wait for image replication to be ready before returning. Default to be false because it's a pretty time consuming process. 

Require at least one available region when updating the `replica_regions` field. 

Add more test cases for image replication.

## ✔️ How to Test

```
make PKG_NAME=linode/image int-test
```

Manual Test:
1. In a sandbox environment, run the following config to perform a image replication
```
resource "linode_instance" "foobar" {
    label = "test-tf-instance"
    type = "g6-standard-1"
    region = "us-east"
    image = "linode/alpine3.19"
    booted = true
}

resource "linode_image" "foobar" {
    linode_id = "${linode_instance.foobar.id}"
    disk_id = "${linode_instance.foobar.disk.0.id}"
    label = "test-gen2-image"
    description = "descriptive text"
    tags = ["test"]
    replica_regions = ["us-east", "us-central"]
    wait_for_replications = true
}
```
2. Observe that image is successfully replicated in 2 regions.
3. Change the replica_regions to use a new region
```
resource "linode_image" "foobar" {
    linode_id = "${linode_instance.foobar.id}"
    disk_id = "${linode_instance.foobar.disk.0.id}"
    label = "test-gen2-image"
    description = "descriptive text"
    tags = ["test"]
    replica_regions = ["eu-west"]
    wait_for_replications = true
}
```
4. Expect to see an error.
5. Change around the config to find any edge case.
